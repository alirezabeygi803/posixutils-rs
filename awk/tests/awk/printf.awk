BEGIN {
    printf "Hello, World!\n"
    printf "Integer: %d\n", 42
    printf "Floating-point: %f\n", 3.14159
    name = "Alice"
    printf "Hello, %s!\n", name
    age = 30
    height = 5.9
    printf "Name: %s, Age: %d, Height: %.1f\n", name, age, height
    printf "Left-justified: %-10s|\n", "left"
    printf "Right-justified: %10s|\n", "right"
    printf "Zero-padded: %05d\n", 7
    printf "Hexadecimal: %x\n", 255
    printf "Octal: %o\n", 255
    printf "Scientific notation: %e\n", 12345.6789
    printf "Percentage: %%\n"
    printf "Newline: \\n, Tab: \\t\n"
    printf "Large integer: %d\n", 1234567890123456789
    printf "Negative integer: %d\n", -42
    printf "Negative floating-point: %f\n", -3.14159
    printf "Precision: %.2f\n", 3.14159
    printf "Width and precision: %10.2f\n", 3.14159
    printf "Special characters: %s\n", "Hello, \tWorld!\n"
    printf "Empty string: '%s'\n", ""
}