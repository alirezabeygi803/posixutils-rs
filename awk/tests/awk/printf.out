Hello, World!
Integer: 42
Floating-point: 3.141590
Hello, Alice!
Name: Alice, Age: 30, Height: 5.9
Left-justified: left      |
Right-justified:      right|
Zero-padded: 00007
Hexadecimal: ff
Octal: 377
Scientific notation: 1.234568e+04
Percentage: %
Newline: \n, Tab: \t
Large integer: 1234567890123456768
Negative integer: -42
Negative floating-point: -3.141590
Precision: 3.14
Width and precision:       3.14
Special characters: Hello, 	World!

Empty string: ''
